index.bs

```diff
@@ -403,107 +410,100 @@ these steps:
 
 </div>
 
-## The {{FileSystemWriter}} interface ## {#api-filesystemwriter}
+## The {{FileSystemWritableFileStream}} interface ## {#api-filesystemwritablefilestream}
 
 <xmp class=idl>
 [Exposed=(Window,Worker), SecureContext]
-interface FileSystemWriter {
+interface FileSystemWritableFileStream : WritableStream {
   Promise<void> write(unsigned long long position, (BufferSource or Blob or USVString) data);
-  WritableStream asWritableStream();
   Promise<void> truncate(unsigned long long size);
   Promise<void> close();
 };
 </xmp>
 
-A {{FileSystemWriter}} has an associated <dfn for=FileSystemWriter>atomic flag</dfn>.
-
-### The {{FileSystemWriter/write()}} method ### {#api-filesystemwriter-write}
-
-<div class="note domintro">
-  : await |writer| . {{FileSystemWriter/write()|write}}(|position|, |data|)
-  :: Writes the content of |data| into the file associated with |writer| at position |position|.
-     If |position| is past the end of the file writing will fail and this method rejects.
-
-     There are no guarantees that the data written to the file using this method is actually
-     reflected in the file on disk until the {{FileSystemWriter/close()}}
-     method is called. Specifically when |writer|'s
-     [=atomic flag=] is set, the user agent will ensure that no changes are written to the actual
-     file until one of those methods is called.
-</div>
+A {{FileSystemWritableFileStream}} object is a {{WritableStream}} object with additional methods to operate
+on a single file on disk.
 
-<div algorithm>
-The <dfn method for=FileSystemWriter>write(|position|, |data|)</dfn> method, when invoked, must run
-these steps:
+The {{FileSystemWritableFileStream}} can be initialized with a |position|. When obtaining a writer through
+{{WritableStream/getWriter()|WritableStream's getWriter()}}, and using the returned
+{{WritableStreamDefaultWriter/write()|WritableStreamDefaultWriter's write()}} method, this position will be
+advanced based on the number of bytes written through this stream object.
 
-1. TODO
+{{WritableStream/getWriter()|WritableStream's getWriter()}} method returns a raw {{WritableStreamDefaultWriter}}.
```

Comment:
What does "raw" mean? How is the writer initialized?

Reply:
changed the wording.